sealtk/gui/Enums.hpp

```diff
@@ -0,0 +1,45 @@
+/* This file is part of SEAL-TK, and is distributed under the OSI-approved BSD
+ * 3-Clause License. See top-level LICENSE file or
+ * https://github.com/Kitware/seal-tk/blob/master/LICENSE for details. */
+
+#ifndef sealtk_gui_Enums_hpp
+#define sealtk_gui_Enums_hpp
+
+#include <QObject>
+#include <qtGlobal.h>
```

Comment:
Pedantic: s.b.:

```c++
#include <qtGlobal.h>

#include <QObject>
```

...except I don't think you actually need either header. I think just `#include <QMetaObject>` is enough?

Reply:
The [QObject documentation](https://doc.qt.io/qt-5/qobject.html#macros) claims that `Q_NAMESPACE` comes from `QObject`. `qtGlobal.h` was indeed cargo-culted from elsewhere.

Reply:
Curious... the documentation is indeed in `qobject.cpp`, but the macro actually comes from `qobjectdefs.h` which apparently corresponds to `QMetaObject`. (I wonder if that's even intended?)

Reply:
In general, I try to go with what the documentation says unless there's a good reason not to. I have removed `qtGlobal.h` and am leaving `QObject` as it is.

Reply:
> [...] unless there's a good reason not to.

Header      |  Raw   | sloccount | time
----------- | ------ | --------- | ----
QObject     | 72,779 |    51,766 | 0.8s
QMetaObject | 28,123 |    19,373 | 0.4s

...? I'd as soon not forcibly drag in all of the extra junk from QObject just for some enumerations.

Reply:
A 0.4s difference in build time on a project that takes ~15s to build does not constitute a "good reason" to go against what is officially supported. If using it from `QMetaObject` is supported, then I will change it, otherwise I will leave it as it is.

Reply:
Because the Qt mailing list did not come to a consensus on whether `Q_NAMESPACE` comes from `QMetaObject` or `QObject`, and because it's listed in the `QObject` documentation, I am going with what the documentation says by default.